src/java/org/apache/cassandra/db/filter/RowFilter.java

```diff
@@ -529,7 +529,7 @@ protected ByteBuffer getValue(TableMetadata metadata, DecoratedKey partitionKey,
                     return row.clustering().bufferAt(column.position());
                 default:
                     Cell<?> cell = row.getCell(column);
-                    return cell == null ? null : cell.buffer();
+                    return Cell.getValidCellBuffer(cell, nowInSec);
```

Comment:
I think we should just replace this w/ `return cell == null || cell.isTombstone() || !cell.isLive(nowInSec) ? null : cell.buffer();` and remove `Cell.getValidCellBuffer()`. What we're doing here is IMO specific to filtering, and at the very least the buffer in a tombstoned cell isn't "invalid"...it's just the buffer value when the cell is tombstoned.

Reply:
how about we move this function to this class rather than `Cell`?

Reply:
sounds good to me

Reply:
I'm basically +1 after this change and the other little nits.